### ISSUE TITLE
SDK potentially looking at the wrong return receipt when a contract calls another contract

### ISSUE THREAD

**Issue by mohammadfawaz** [FuelLabs/fuels-rs#609]
Reproducer here: https://github.com/FossilFrank/fuel-issue

Run `forc build` for the exchange, then `forc test` for the router. There's 2 test cases: one basic one that should pass, and another one that has an internal call that fails.

Comparing receipts: Left is bad (i.e.  `get_pool_info()` is called) , right is good (i.e. `get_pool_info()` is not called).
![image](https://user-images.githubusercontent.com/59666792/193854377-00383838-be88-4514-abc3-f4fe5eb28163.png)

To summarize, we have a contract method `A` that calls another contract method `B`. The return value from `A` in the SDK seems to be overriden by the return value from `B` so the SDK reports the wrong values in `response.value`

If it turns out that this is *not* a problem with the SDK and it is indeed an issue in the compiler, please let me know!

It's worth checking if this is an issue in `fuels-ts` as well.
